src/runtime.js

```diff
@@ -118,6 +118,11 @@ function adoptShared(global, callback) {
     win: global,
   };
 
+  function emptyService() {
+    // All services need to resolve to an object.
+    return {};
```

Comment:
Is this in hot path? If so, creating only once as a constant outside might be beneficial.